pkg/middlewares/aws_instance_block_device.go

```diff
@@ -28,6 +32,110 @@ func (a AwsInstanceBlockDeviceResourceMapper) Execute(remoteResources, resources
 			continue
 		}
 
+		if resource.IsRefactoredResource("aws_instance") {
+			instance, _ := stateRes.(*resource.AbstractResource)
+
+			if rootBlockDevice, exist := instance.Attrs.Get("root_block_device"); exist {
+				for _, rootBlock := range rootBlockDevice.([]interface{}) {
+					rootBlock := rootBlock.(map[string]interface{})
+					logrus.WithFields(logrus.Fields{
+						"volume":   rootBlock["volume_id"],
+						"instance": instance.TerraformId(),
+					}).Debug("Creating aws_ebs_volume from aws_instance.root_block_device")
+					data := map[string]interface{}{
+						"availability_zone": func() interface{} {
+							if val, ok := instance.Attrs.Get("availability_zone"); ok {
+								return val
+							}
+							return nil
+						}(),
+						"encrypted":            rootBlock["encrypted"],
+						"id":                   rootBlock["volume_id"],
+						"iops":                 rootBlock["iops"],
+						"kms_key_id":           rootBlock["kms_key_id"],
+						"size":                 rootBlock["volume_size"],
+						"type":                 rootBlock["volume_type"],
+						"multi_attach_enabled": false,
+						"tags": func() interface{} {
+							if val, ok := instance.Attrs.Get("volume_tags"); ok {
+								return val
+							}
+							return nil
+						}(),
+					}
+					ctyVal, err := a.resourceFactory.CreateResource(data, "aws_ebs_volume")
+					if err != nil {
+						return err
+					}
+
+					schema, exist := a.resourceSchemaRepository.GetSchema("aws_ebs_volume")
```

Comment:
We should encapsulate this logic in `resourceFactory` IMHO. We may introduce a `CreateAbstractResource` should return an `AbstractResource` without any need to pass by a ctyVal.

`data` can be used as it as an `resource.Attributes`, calling `ToResourceAttributes` there is useless